src/qt/masternodelist.cpp

```diff
@@ -500,8 +497,8 @@ void MasternodeList::ShowQRCode(std::string strAlias) {
     strHTML += "<b>" + tr("IP") +               ": </b>" + GUIUtil::HtmlEscape(strIP) + "<br>";
     if (fFound) {
         strHTML += "<b>" + tr("Protocol") +     ": </b>" + QString::number(mn.nProtocolVersion) + "<br>";
-        strHTML += "<b>" + tr("Version") +      ": </b>" + (mn.lastPing.nDaemonVersion > DEFAULT_DAEMON_VERSION ? GUIUtil::HtmlEscape(FormatVersion(mn.lastPing.nDaemonVersion)) : tr("Unknown")) + "<br>";
-        strHTML += "<b>" + tr("Sentinel") +     ": </b>" + (mn.lastPing.nSentinelVersion > DEFAULT_SENTINEL_VERSION ? GUIUtil::HtmlEscape(SafeIntVersionToString(mn.lastPing.nSentinelVersion)) : tr("Unknown")) + "<br>";
+        strHTML += "<b>" + tr("Version") +      ": </b>" + GUIUtil::HtmlEscape(mn.lastPing.GetDaemonString()) + "<br>";
+        strHTML += "<b>" + tr("Sentinel") +     ": </b>" + GUIUtil::HtmlEscape(mn.lastPing.GetSentinelString()) + "<br>";
```

Comment:
New code won't produce the same results - unlike RPC, GUI uses `tr` (i.e. translation) for the "Unknown" string.

Reply:
Yep, I see your point. But I'd really like to try and find a solution that doesn't duplicate the logic both here and in the Get<X>String methods. Is there a way of providing translation for "Unknown" in the code for (e.g.) GetDaemonString instead?

Reply:
The problem is that it's common to see translated strings in GUI but not in RPC, that's why it was implemented this way. I don't see how to reuse it cleanly tbh, maybe only smth like
```diff
diff --git a/src/qt/masternodelist.cpp b/src/qt/masternodelist.cpp
index 266a14335..e2a42df4a 100644
--- a/src/qt/masternodelist.cpp
+++ b/src/qt/masternodelist.cpp
@@ -492,6 +492,9 @@ void MasternodeList::ShowQRCode(std::string strAlias) {
     // Title above QR-Code
     QString strQRCodeTitle = tr("Masternode Private Key");
 
+    std::string strDaemon = mn.lastPing.GetDaemonString();
+    std::string strSentinel = mn.lastPing.GetSentinelString();
+
     // Create dialog text as HTML
     QString strHTML = "<html><font face='verdana, arial, helvetica, sans-serif'>";
     strHTML += "<b>" + tr("Alias") +            ": </b>" + GUIUtil::HtmlEscape(strAlias) + "<br>";
@@ -500,8 +503,8 @@ void MasternodeList::ShowQRCode(std::string strAlias) {
     strHTML += "<b>" + tr("IP") +               ": </b>" + GUIUtil::HtmlEscape(strIP) + "<br>";
     if (fFound) {
         strHTML += "<b>" + tr("Protocol") +     ": </b>" + QString::number(mn.nProtocolVersion) + "<br>";
-        strHTML += "<b>" + tr("Version") +      ": </b>" + (mn.lastPing.nDaemonVersion > DEFAULT_DAEMON_VERSION ? GUIUtil::HtmlEscape(FormatVersion(mn.lastPing.nDaemonVersion)) : tr("Unknown")) + "<br>";
-        strHTML += "<b>" + tr("Sentinel") +     ": </b>" + (mn.lastPing.nSentinelVersion > DEFAULT_SENTINEL_VERSION ? GUIUtil::HtmlEscape(SafeIntVersionToString(mn.lastPing.nSentinelVersion)) : tr("Unknown")) + "<br>";
+        strHTML += "<b>" + tr("Version") +      ": </b>" + (strDaemon == "Unknown" ? tr("Unknown") : strDaemon.c_str()) + "<br>";
+        strHTML += "<b>" + tr("Sentinel") +     ": </b>" + (strDaemon == "Unknown" ? tr("Unknown") : strSentinel.c_str()) + "<br>";
         strHTML += "<b>" + tr("Status") +       ": </b>" + GUIUtil::HtmlEscape(CMasternode::StateToString(mn.nActiveState)) + "<br>";
         strHTML += "<b>" + tr("Payee") +        ": </b>" + GUIUtil::HtmlEscape(CBitcoinAddress(mn.pubKeyCollateralAddress.GetID()).ToString()) + "<br>";
         strHTML += "<b>" + tr("Active") +       ": </b>" + GUIUtil::HtmlEscape(DurationToDHMS(mn.lastPing.sigTime - mn.sigTime)) + "<br>";
```

Reply:
@UdjinM6 I've been thinking a bit more on this... is "Unknown" really necessary at all? Can we use a "0" version string, or an empty string instead? Just seems weird that we have these special processing rules based on the string content.

That said, I fully understand the `tr()` for Qt only, but if we're putting in a either version _or_ "Unknown", then it's obviously for human consumption, so why not make it more consistent and remove these conditionals?

Reply:
Agree, maybe we should get rid of "Unknown" (less strings to translate, yay! :) ) but not in this (refactoring) PR imo.